File: C#Basics/EntryExam/01. Triangle/Triangle.cs
using System;

class Triangle
{
    static void Main()
    {
        double Ax = double.Parse(Console.ReadLine());
        double Ay = double.Parse(Console.ReadLine());
        double Bx = double.Parse(Console.ReadLine());
        double By = double.Parse(Console.ReadLine());
        double Cx = double.Parse(Console.ReadLine());
        double Cy = double.Parse(Console.ReadLine());

        double a = Math.Sqrt((Cx - Bx) * (Cx - Bx) + (Cy - By) * (Cy - By));
        double b = Math.Sqrt((Ax - Cx) * (Ax - Cx) + (Ay - Cy) * (Ay - Cy));
        double c = Math.Sqrt((Bx - Ax) * (Bx - Ax) + (By - Ay) * (By - Ay));

        bool canFormTriangle = a + b > c && b + c > a && a + c > b;

        double p = (a + b + c) / 2; //Half perimeter
        double area = Math.Sqrt(p * (p - a) * (p - b) * (p - c));

        if (canFormTriangle)
        {
            Console.WriteLine("Yes");
            Console.WriteLine("{0:0.00}", area);
        }
        else
        {
            Console.WriteLine("No");
            Console.WriteLine("{0:0.00}", c);
        }
    }
}

File: ObjectOrientedProgramming/10. OOPInJava/src/pkg10/oopinjava/geometry/Triangle.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package pkg10.oopinjava.geometry;

public class Triangle extends PlaneShape {

    public Triangle(Vertix2D firstVertix, Vertix2D secondVertix, Vertix2D thirdVertix) {
        super(new Vertix2D[]{firstVertix, secondVertix, thirdVertix});
    }

    @Override
    public double getPerimeter() {
        Vertix2D a = (Vertix2D) this.getVertices()[0];
        Vertix2D b = (Vertix2D) this.getVertices()[1];
        Vertix2D c = (Vertix2D) this.getVertices()[2];
        
        double ab = Math.sqrt((b.getX() - a.getX()) * (b.getX() - a.getX()) + (b.getY() - a.getY()) * (b.getY() - a.getY()));
        double bc = Math.sqrt((c.getX() - b.getX()) * (c.getX() - b.getX()) + (c.getY() - b.getY()) * (c.getY() - b.getY()));
        double ac = Math.sqrt((c.getX() - a.getX()) * (c.getX() - a.getX()) + (c.getY() - a.getY()) * (c.getY() - a.getY()));
        
        double perimeter = ab + bc + ac;
        
        return perimeter;
    }

    @Override
    public double getArea() {
        Vertix2D a = (Vertix2D) this.getVertices()[0];
        Vertix2D b = (Vertix2D) this.getVertices()[1];
        Vertix2D c = (Vertix2D) this.getVertices()[2];
        double area = Math.abs((a.getX() * (b.getY() - c.getY()) + b.getX() * (c.getY() - a.getY()) + c.getX() * (a.getY() - b.getY())) / 2);

        return area;
    }
}
